go.mod

```diff
@@ -2,6 +2,8 @@ module github.com/ONSdigital/dp-dataset-exporter
 
 go 1.22
 
+replace github.com/ONSdigital/dp-s3 => github.com/ONSdigital/dp-s3/v2 v2.1.0
```

Comment:
You don't need a replace statement for this. You need to replace all the code references and then change the go.mod file to require dp-s3/v2